CODE-QUALITY_CHECKS.md

```diff
@@ -0,0 +1,244 @@
+with your project repository, providing you with feedback in two forms:
+
+* Badges (at the top of [README.md](README.md))
+* Pull request checks
+
+Unfortunately these badges and checks need to be (re)configured
+every time you or GitHub Classroom makes a new repository. This is
+
+* Every time GitHub Classroom creates a new repo from our starter code
+* Every time a group forks or copies a repository from one iteration to another
+
+Here's a checklist of actions you'll need to take for each
+new repository:
+
+* [ ] Fix BetterCodeHub badge
+* [ ] Fix LGTM badge(s)
+* [ ] Enable PR checks for BetterCodeHub
+* [ ] Enable PR checks for LGTM
+
+This document includes:
+
+* Descriptions of the analysis tools we're currently using
+* Info on how to configure the badges
+* Info on how to enable the pull request checks
+
+## What analysis tools are we using and why?
+
+> Feel free to skip this and jump down to the sections on how to
+> configure things if you wish. But you might want to come back to
+> it, especially if you're unclear about what the feedback
+> you're getting from these services means.
+
+At the end of the day, nothing really replaces a quality code review
+process. These help ensure that everyone on the team is on the same
+page and knows what's happening with the project, and helps ensure that
+the project is up to the team's standards. There are, however, things
+that it's just easy to overlook, and things no one on the team is
+actually aware of.
+
+Code analysis tools can help make sure that at least a variety of
+automatic checks happen on each pull request (PR). Then if a PR would
+introduce a problem or in some other way "lower" the quality of the
+code, everyone would know it and have a chance to address it before
+that PR is merged in.
+
+Badges also provide a nice visual indication of the health of the code
+base. This can be useful for getting a quick sense of the state of the
+project, as well as "advertising" that you care about quality and are
+making an effort to maintain the quality of your codebase.
+
+There are quite a few code analysis tools out there; we're using two
+that, together, seem to cover a lot of the major issues in ways that
+are pretty "user friendly": BetterCodeHub.com and LGTM.com. We make no
+claims that these are the "best", however, and feel free to
+explore others, sharing cool things that you discover.
+
+Some other services:
+
+* CodeScene.io provides quite sophisticated analyses of your codebase,
+  but it's not trivial to make sense of the huge amount of information
+  that it provides.
+* Codacy.com is cool in that it understands a _lot_ of languages
+  and notations, but it's defaults can be quite "noisy" and it's not
+  trivial to reconfigure.
+
+### BetterCodeHub
+
+BetterCodeHub.com provides a "big picture" analysis of your codebase,
+checking 10 particular properties. Many of these are different versions
+of "keep things simple", although there are also architecture-level
```

Comment:
```suggestion
of "keep things simple," although there are also architecture-level
```

Reply:
You're technically right, but I find that grammar rule ludicrous. There's a reason not to interleave grouping symbols in programming languages, and that really ought to apply to English as well. 😜

Reply:
For sure, I think it's dumb too 🤷